zebra/kernel_netlink.c

```diff
@@ -697,6 +697,204 @@ static void netlink_parse_extended_ack(struct nlmsghdr *h)
 	}
 }
 
+/*
+ * netlink_send_msg - send a netlink message of a certain size.
+ *
+ * Returns -1 on error. Otherwise, it returns the number of bytes sent.
+ */
+static int netlink_send_msg(const struct nlsock *nl, void *buf, size_t buflen)
+{
+	struct sockaddr_nl snl;
+	struct iovec iov;
+	struct msghdr msg;
+	int status, save_errno = 0;
+
+	memset(&snl, 0, sizeof(snl));
```

Comment:
**Non blocking comment**

How about using `snl = {}`, `msg = {}` and `iov = {}` instead of `memset`ing?

Reply:
Looks good. I changed this.